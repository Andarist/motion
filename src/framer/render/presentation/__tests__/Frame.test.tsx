import * as React from "react"
import { Frame, RenderTarget, executeInRenderEnvironment } from "../../"
import { mount, ReactWrapper } from "enzyme"

function child(wrapper: ReactWrapper) {
    return wrapper.childAt(0).childAt(0)
}

function childDiv(wrapper: ReactWrapper) {
    return child(wrapper).childAt(0)
}

test("should resize a child frame with RenderTarget.canvas", () => {
    Frame.prototype.render = jest.fn(Frame.prototype.render)
    const wrapper = executeInRenderEnvironment({ target: RenderTarget.canvas }, () => {
        const w = mount(
            <Frame width={50} height={50}>
                <Frame width={10} height={10} right={5} />
            </Frame>
        )
        expect(childDiv(w).prop("style")).toMatchSnapshot()
        expect(Frame.prototype.render).toHaveBeenCalledTimes(2)
        w.setProps({ width: 100 })
        return w
    })
    expect(childDiv(wrapper).prop("style")).toMatchSnapshot()
    const c = child(wrapper).instance() as Frame
    expect(c.element!.style).toMatchSnapshot()
    expect(Frame.prototype.render).toHaveBeenCalledTimes(4)
})

test("should resize a child frame with CanvasMode.live without rerendering", () => {
    Frame.prototype.render = jest.fn(Frame.prototype.render)
    const wrapper = executeInRenderEnvironment({ target: RenderTarget.preview }, () => {
        const w = mount(
            <Frame width={50} height={50} name={"parent"}>
                <Frame width={10} height={10} right={5} name={"child"} />
            </Frame>
        )
        expect(childDiv(w).prop("style")).toMatchSnapshot()
        expect(Frame.prototype.render).toHaveBeenCalledTimes(2)
        w.setProps({ width: 100 })
        return w
    })
    const c = child(wrapper).instance() as Frame
    expect(c.element!.style).toMatchSnapshot()
    expect(Frame.prototype.render).toHaveBeenCalledTimes(3)
})
