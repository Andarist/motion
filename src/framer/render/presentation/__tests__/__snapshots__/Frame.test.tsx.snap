// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should resize a child frame with CanvasMode.live without rerendering 1`] = `
Object {
  "WebkitBackfaceVisibility": undefined,
  "WebkitPerspective": undefined,
  "WebkitTransformOrigin": "50% 50%",
  "backfaceVisibility": undefined,
  "backgroundColor": "rgba(0, 170, 255, 0.3)",
  "display": "block",
  "height": "10px",
  "perspective": undefined,
  "pointerEvents": undefined,
  "position": "absolute",
  "transform": "
            translate(35px, 20px)
            scale(1, 1)
            skew(0deg,0deg)
            skewX(0deg)
            skewY(0deg)
            rotate(0.0000deg)",
  "transformOrigin": "50% 50%",
  "userSelect": "none",
  "width": "10px",
  "willChange": "transform",
}
`;

exports[`should resize a child frame with CanvasMode.live without rerendering 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "position",
  "2": "width",
  "3": "height",
  "4": "transform",
  "5": "background-color",
  "WebkitBackfaceVisibility": undefined,
  "WebkitPerspective": undefined,
  "WebkitTransformOrigin": "50% 50%",
  "_importants": Object {
    "background-color": undefined,
    "display": undefined,
    "height": undefined,
    "position": undefined,
    "transform": undefined,
    "width": undefined,
  },
  "_length": 6,
  "_onChange": [Function],
  "_values": Object {
    "background-color": "rgba(0, 170, 255, 0.3)",
    "display": "block",
    "height": "10px",
    "position": "absolute",
    "transform": "
            translate(85px, 20px)
            scale(1, 1)
            skew(0deg,0deg)
            skewX(0deg)
            skewY(0deg)
            rotate(0.0000deg)",
    "width": "10px",
  },
  "backfaceVisibility": undefined,
  "perspective": undefined,
  "transformOrigin": "50% 50%",
  "userSelect": "none",
  "willChange": "transform",
}
`;

exports[`should resize a child frame with RenderTarget.canvas 1`] = `
Object {
  "WebkitBackfaceVisibility": undefined,
  "WebkitPerspective": undefined,
  "WebkitTransformOrigin": "50% 50%",
  "backfaceVisibility": undefined,
  "backgroundColor": "rgba(0, 170, 255, 0.3)",
  "display": "block",
  "height": "10px",
  "perspective": undefined,
  "pointerEvents": undefined,
  "position": "absolute",
  "transform": "
            translate(35px, 20px)
            scale(1, 1)
            skew(0deg,0deg)
            skewX(0deg)
            skewY(0deg)
            rotate(0.0000deg)",
  "transformOrigin": "50% 50%",
  "userSelect": "none",
  "width": "10px",
  "willChange": "transform",
}
`;

exports[`should resize a child frame with RenderTarget.canvas 2`] = `
Object {
  "WebkitBackfaceVisibility": undefined,
  "WebkitPerspective": undefined,
  "WebkitTransformOrigin": "50% 50%",
  "backfaceVisibility": undefined,
  "backgroundColor": "rgba(0, 170, 255, 0.3)",
  "display": "block",
  "height": "10px",
  "perspective": undefined,
  "pointerEvents": undefined,
  "position": "absolute",
  "transform": "
            translate(85px, 20px)
            scale(1, 1)
            skew(0deg,0deg)
            skewX(0deg)
            skewY(0deg)
            rotate(0.0000deg)",
  "transformOrigin": "50% 50%",
  "userSelect": "none",
  "width": "10px",
  "willChange": "transform",
}
`;

exports[`should resize a child frame with RenderTarget.canvas 3`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "position",
  "2": "width",
  "3": "height",
  "4": "transform",
  "5": "background-color",
  "WebkitBackfaceVisibility": "",
  "WebkitPerspective": "",
  "WebkitTransformOrigin": "50% 50%",
  "_importants": Object {
    "background-color": undefined,
    "display": undefined,
    "height": undefined,
    "position": undefined,
    "transform": undefined,
    "width": undefined,
  },
  "_length": 6,
  "_onChange": [Function],
  "_values": Object {
    "background-color": "rgba(0, 170, 255, 0.3)",
    "display": "block",
    "height": "10px",
    "position": "absolute",
    "transform": "translate(85px, 20px)
            scale(1, 1)
            skew(0deg,0deg)
            skewX(0deg)
            skewY(0deg)
            rotate(0.0000deg)",
    "width": "10px",
  },
  "backfaceVisibility": "",
  "perspective": "",
  "transformOrigin": "50% 50%",
  "userSelect": "none",
  "willChange": "transform",
}
`;
